consensus/src/sync/live/block_queue/block_request_component.rs

```diff
@@ -318,6 +318,10 @@ impl<N: Network> BlockRequestComponent<N> {
         self.pending_requests.contains(target_block_hash)
     }
 
+    pub fn has_no_pending_requests(&self) -> bool {
+        self.pending_requests.is_empty()
```

Comment:
```suggestion
    pub fn has_pending_requests(&self) -> bool {
        !self.pending_requests.is_empty()
```
Avoid negative function names as they lead to double negation if we need to invert them.